generator/internal/descriptor_utils.cc

```diff
@@ -210,13 +211,79 @@ std::string DefaultIdempotencyFromHttpOperation(
       std::exit(1);
   }
 }
+
+std::string FormatClassCommentsFromServiceComments(
+    google::protobuf::ServiceDescriptor const& service) {
+  google::protobuf::SourceLocation service_source_location;
+  if (service.GetSourceLocation(&service_source_location) &&
+      !service_source_location.leading_comments.empty()) {
+    return absl::StrReplaceAll(
```

Comment:
This 7-line one liner is a bit difficult to read. There's lots of asterisks and newlines around, but it's difficult to grok what's happening. Consider breaking this up into multiple lines so that the steps transforming the service comments into C++ comments are easier to follow.